In [1]:
# connect to db
# Import the load_dotenv function from the dotenv module
from dotenv import load_dotenv
import os

# Load environment variables from .env file
load_dotenv()

# Retrieve connection parameters from environment variables
username = os.getenv('username')
password = os.getenv('password')
host = os.getenv('host')
port = os.getenv('port')
database = os.getenv('database')

# Form the connection string for the SQL magic command
connection_string = f"mysql+pymysql://{username}:{password}@{host}:{port}/{database}"

# Load SQL magic if not already loaded
%load_ext sql

# Connect to the database using the connection string
%sql $connection_string



## Business Question:
Which companies have unusually high cash flow per share compared to their total debt, possibly indicating cash flow manipulation or other irregular financial reporting?

In [35]:
%%sql
WITH HighCashDebtSymbols AS (
    SELECT symbol, 
           cashFlowPerShareAnnual,
           `totalDebt/TotalEquityAnnual`,
           (cashFlowPerShareAnnual / `totalDebt/TotalEquityAnnual`) AS cash_debt_ratio
    FROM basic_financials
)
SELECT symbol, cashFlowPerShareAnnual, `totalDebt/TotalEquityAnnual`, cash_debt_ratio
FROM HighCashDebtSymbols
WHERE cash_debt_ratio > 2
ORDER BY cash_debt_ratio DESC
LIMIT 10;


 * mysql+pymysql://admin:***@mayah-bosworth-sql-project.cmhtgvzs0rf5.us-east-1.rds.amazonaws.com:3306/mayah_bosworth_sql_project
8 rows affected.


[('XOM', 8.4236, 0.2036, 41.37328094302554),
 ('JLL', 8.1857, 0.3496, 23.414473684210527),
 ('AMZN', 3.1029, 0.4149, 7.478669558929862),
 ('DIS', 2.7496, 0.4701, 5.848968304616039),
 ('TPR', 3.4785, 0.7302, 4.763763352506163),
 ('MRK', 3.6115, 0.9328, 3.8716766723842198),
 ('AAPL', 6.4041, 1.804, 3.5499445676274943),
 ('JPM', 4.5101, 1.866, 2.416988210075027)]

## Insights:
- Cash-to-debt ratios vary significantly, with XOM having the highest at 41.37 and JPM the lowest at 2.42.
- XOM shows strong liquidity, while JPM indicates relatively lower liquidity.
- Symbols like JLL, AMZN, and DIS also have high cash-to-debt ratios, indicating favorable liquidity positions.

## Recommendations:
- XOM should maintain prudent financial management and consider strategic investments or debt reduction.
- JPM may optimize cash flow management or explore debt restructuring.
- Evaluate reasons for variations and tailor financial strategies accordingly.

## Predictions:
- Investors will monitor liquidity, especially for high-debt companies.
- Strong liquidity may attract investors, while weaker liquidity may face scrutiny.
- Continued monitoring and proactive financial management are crucial for navigating uncertainties and sustaining growth.

## Business Question:
Which companies demonstrate a low quick ratio but high inventory turnover, suggesting potential overstatement of inventory or other financial discrepancies?

In [36]:
%%sql
WITH LowLiquidityHighInventorySymbols AS (
    SELECT symbol,
           quickRatioAnnual,
           inventoryTurnoverAnnual,
           (quickRatioAnnual / inventoryTurnoverAnnual) AS liquidity_inventory_balance
    FROM basic_financials
)
SELECT symbol, quickRatioAnnual, inventoryTurnoverAnnual, liquidity_inventory_balance
FROM LowLiquidityHighInventorySymbols
WHERE quickRatioAnnual < 1 AND inventoryTurnoverAnnual > 5 
ORDER BY liquidity_inventory_balance;




 * mysql+pymysql://admin:***@mayah-bosworth-sql-project.cmhtgvzs0rf5.us-east-1.rds.amazonaws.com:3306/mayah_bosworth_sql_project
7 rows affected.


[('AAPL', 0.9444, 37.9777, 0.024867224713450264),
 ('DIS', 0.9891, 31.9574, 0.030950577956905128),
 ('GS', 0.3568, 9.13935, 0.039039975490598344),
 ('AMZN', 0.8103, 8.9996, 0.09003733499266635),
 ('JLL', 0.9757, 9.13935, 0.10675813925497983),
 ('C', 0.9823999999999999, 9.13935, 0.10749123296514522),
 ('JPM', 0.9823999999999999, 9.13935, 0.10749123296514522)]

## Insights:
- The liquidity-to-inventory balance varies among symbols.
- Symbols like GS have lower balances, indicating potential liquidity challenges.
- Symbols like AMZN, JLL, C, and JPM show higher balances, suggesting better liquidity management.

## Recommendations:
- For low-balance companies like GS, optimize inventory management to improve liquidity.
- High-balance companies should monitor turnover rates for efficient working capital management.

## Prediction:
- Stakeholders will closely monitor liquidity-to-inventory balances, especially in inventory-heavy industries.
- Optimal balances will position companies to navigate market fluctuations and capitalize on growth opportunities.
- Maintaining a healthy balance between liquidity and inventory turnover is vital for sustained financial stability and success.